src/app/pages/onboarding/set-password/components/password-field.tsx

```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react';
+import { FiEye, FiEyeOff } from 'react-icons/fi';
+import { useField } from 'formik';
+import { Box, color, Flex, Input } from '@stacks/ui';
+
+import { ValidatedPassword } from '@app/common/validation/validate-password';
+import { Caption } from '@app/components/typography';
+import { OnboardingSelectors } from '@tests/integration/onboarding/onboarding.selectors';
+
+import { PasswordStrengthIndicator } from './password-strength-indicator';
+import { getIndicatorsOfPasswordStrength } from './password-field.utils';
+
+interface PasswordFieldProps {
+  strengthResult: ValidatedPassword;
+}
+export function PasswordField({ strengthResult }: PasswordFieldProps) {
+  const [field] = useField('password');
+  const [showPassword, setShowPassword] = useState(false);
+
+  const { strengthColor, strengthText } = getIndicatorsOfPasswordStrength(strengthResult);
+
+  return (
+    <>
+      <Box position="relative">
+        <Input
+          _focus={{ border: '2px solid #5546FF' }}
+          autoFocus
+          border="2px solid"
+          data-testid={OnboardingSelectors.NewPasswordInput}
+          height="64px"
+          key="password-input"
+          placeholder="Set a password"
+          type={showPassword ? 'text' : 'password'}
+          {...field}
+        />
+        <Box
+          as={showPassword ? FiEyeOff : FiEye}
+          color={color('text-title')}
+          onClick={field.value ? () => setShowPassword(!showPassword) : undefined}
+          position="absolute"
+          right="base"
+          size="20px"
+          top="22px"
+          transform="matrix(-1, 0, 0, 1, 0, 0)"
+        />
```

Comment:
I like the toggle, but this is bad accessibility. As an interactive element, it should be a `button`, rather than a `div`. 

Consider that: the cursor should turn into a hand, it should be triggerable via focus state etc

Also, what is the matrix required for? 🤔

Reply:
Ok, I can change it to a button. Fer had the transform on there to flip the direction of the FiEyeOff icon.